src/backend/joanie/tests/core/test_models_order.py

```diff
@@ -85,34 +89,35 @@ def test_models_order_course_in_product_existing():
 
     def test_models_order_state_property(self):
         """
-        Order state property is dynamically computed from `is_canceled` state
-        and related invoice.
+        Order state property is set and related with invoice.
         """
         course = factories.CourseFactory()
         product = factories.ProductFactory(title="Traçabilité", courses=[course])
-        order = factories.OrderFactory(product=product, is_canceled=False)
+        order = factories.OrderFactory(product=product, state=enums.ORDER_STATE_PENDING)
 
         # 1 - By default, an order is `pending``
         self.assertEqual(order.state, enums.ORDER_STATE_PENDING)
 
-        # 2 - When an invoice is linked to the order, its state is `validated`
+        # 2 - When an invoice is linked to the order, and the method validate() is
+        # called its state is `validated`
         InvoiceFactory(order=order, total=order.total)
+        order.validate()
```

Comment:
Should we automatically validate an order on invoice creation/update ?

Reply:
IMO the evolution should more be to automatically create an invoice when we validate the order but this needs to be discussed

Reply:
+1 the order creates the invoice in its transition to be validated.

Reply:
I would leave this update out of this PR with the PR taking into account the balance wdyt? I think both subjects are linked